compiler/rustc_mir/src/transform/normalize_array_len.rs

```diff
@@ -0,0 +1,249 @@
+//! This pass eliminates casting of arrays into slices when their length
+//! is taken using `.len()` method. Handy to preserve information in MIR for const prop
+
+use crate::transform::MirPass;
+use rustc_data_structures::fx::FxIndexMap;
+use rustc_index::vec::IndexVec;
+use rustc_middle::mir::*;
+use rustc_middle::ty::{self, TyCtxt};
+
+pub struct NormalizeArrayLen;
+
+impl<'tcx> MirPass<'tcx> for NormalizeArrayLen {
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+        // if tcx.sess.mir_opt_level() < 3 {
+        //     return;
+        // }
```

Comment:
Why is the commented out?

Reply:
An appropriate opt level can be chosen before landing, but for perf runs it needs to always run.

Reply:
This should be uncommented now that the perf runs have finished